apps/nextjs/src/app/onboarding/onboarding.tsx

```diff
@@ -1,128 +1,46 @@
 "use client";
 
-import { useState } from "react";
+import { useEffect, useRef } from "react";
 
+import { useUser } from "@clerk/nextjs";
 import logger from "@oakai/logger/browser";
-import {
-  OakBox,
-  OakFlex,
-  OakHeading,
-  OakLink,
-  OakP,
-  OakPrimaryButton,
-  OakSpan,
-} from "@oaknational/oak-components";
-import Link from "next/link";
+import { useReloadSession } from "hooks/useReloadSession";
 
-import Button from "@/components/Button";
-import CheckBox from "@/components/CheckBox";
-import SignUpSignInLayout from "@/components/SignUpSignInLayout";
-import TermsContent from "@/components/TermsContent";
+import { AcceptTermsForm } from "@/components/Onboarding/AcceptTermsForm";
+import { LegacyUpgradeNotice } from "@/components/Onboarding/LegacyUpgradeNotice";
 import { trpc } from "@/utils/trpc";
 
 export const OnBoarding = () => {
-  const [dropDownOpen, setDropDownOpen] = useState(true);
-  const [termsAcceptedLocal, setTermsAcceptedLocal] = useState(false);
-  const [privacyAcceptedLocal, setPrivacyAcceptedLocal] = useState(false);
-  const acceptTerms = trpc.auth.acceptTerms.useMutation();
-
-  const handleAcceptTermsOfUse = async () => {
-    try {
-      const response = await acceptTerms.mutateAsync({
-        termsOfUse: new Date(),
-        privacyPolicy: privacyAcceptedLocal ? new Date() : false,
-      });
-
-      if (!response?.acceptedTermsOfUse) {
-        throw new Error("Could not accept terms of use");
+  const { user } = useUser();
+  const reloadSession = useReloadSession();
+  const setDemoStatus = trpc.auth.setDemoStatus.useMutation();
+
+  const userHasAlreadyAcceptedTerms =
+    user?.publicMetadata?.["labs"]?.["isOnboarded"];
+
+  // Edge case: Legacy users have already accepted terms but don't have a demo status
```

Comment:
Sorry being a bit slow here -- where's the condition that they don't have a demo status?

Reply:
Ah just seen -- user won't land on this page otherwise

Reply:
The case is any user who signed up before started capturing the demo status. Unless we can find a sensible representation of their region (maybe posthog?), our only option is to wait until they come back online to set it

We're soon going to have users landing on labs from OWA who don't have all of our metadata, so this concept of upgrading metadata will likely come in useful anyway